### PR TITLE
Drop PHP 5.3 support

### PR DESCRIPTION
**Pull request by rclachance**
This is an alternative to https://github.com/Automattic/WPCOM-Legacy-Redirector/pull/27, dropping support for PHP 5.3 and simplifying the travis config file.

cc @mjangda

### CODE REVIEW

**Comment by mjangda** [comment]
We should probably add 7.0 + 7.1 to the matrix.

@philipjohn thoughts?

File: .travis.yml [hunk 1]
```diff
@@ -6,16 +6,11 @@ notifications:
     on_failure: change
 
 php:
-  - 5.3
   - 5.6
 
 env:
   - WP_VERSION=latest WP_MULTISITE=0
-
-matrix:
```
Comment (mjangda) [inline .travis.yml]: I think we still need the `matrix` to be able to test multsite + non-multisite separately.

Comment (rclations) [reply .travis.yml]: Should we be testing multisite on all PHP versions or only some?

The travis configuration before this commit was [only running multisite on PHP 5.3](https://travis-ci.org/Automattic/WPCOM-Legacy-Redirector/builds/282731430), while this PR [expands it to include PHP 5.6](https://travis-ci.org/Automattic/WPCOM-Legacy-Redirector/builds/282540193?utm_source=github_status&utm_medium=notification) (examples linked)

Comment (mjangda) [reply .travis.yml]: I think it probably makes sense to test on all.

Comment (rclations) [reply .travis.yml]: as best as I understand, Travis will build a matrix using the default configuration set (in this case, all combinations of the `php` and `env` declarations), and the `matrix` declaration can be used to override that using `include` and/or `exclude`.

The documentation on this isn't super clear, but it's available [here](https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix)

However, [Liveblog](https://github.com/Automattic/liveblog/blob/master/.travis.yml) & [Jetpack](https://github.com/Automattic/jetpack/blob/master/.travis.yml) (as examples) both avoid setting a default configuration in favor of defining an include matrix. Happy to switch this over if that's the preferred pattern.

Comment (mjangda) [reply .travis.yml]: Oh, actually, I misunderstood. I think the approach we have here seems fine and works well :)

**Review by mjangda — APPROVED** [review]

**Pull request merged**